pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -371,36 +371,51 @@ func (r *MigPlan) BuildRegistryDC(storage *MigStorage, name, dirName string) (*a
 			Namespace: VeleroNamespace,
 		},
 	}
-	err := r.UpdateRegistryDC(storage, deploymentconfig, name, dirName)
+	err := r.UpdateRegistryDC(storage, deploymentconfig, envVars, name, dirName)
 	return deploymentconfig, err
 }
 
-func buildProxyEnvVars() []kapi.EnvVar {
+// Build up proxy env vars for registry DC
+// Returns empty array if secret isn't found or no configuration exists
+func (r *MigPlan) BuildRegistryProxyEnvVars(client k8sclient.Client) []kapi.EnvVar {
 	envVars := []kapi.EnvVar{}
-	if found, s := Settings.HasProxyVar(settings.HttpProxy); found {
+	secret, err := GetSecret(
+		client,
+		&kapi.ObjectReference{
+			Namespace: VeleroNamespace,
+			Name:      ProxySecret,
+		},
+	)
+	if err != nil {
```

Comment:
The `GetSecret()` will return a `nil` secret when `NotFound`.  Any other error should be returned.